src/community/leiden.c

```diff
@@ -1036,10 +1040,29 @@ igraph_error_t igraph_community_leiden(const igraph_t *graph,
         i_node_weights = (igraph_vector_t*)node_weights;
     }
 
-    /* Perform actual Leiden algorithm */
-    IGRAPH_CHECK(igraph_i_community_leiden(graph, i_edge_weights, i_node_weights,
-                                           resolution_parameter, beta,
-                                           membership, nb_clusters, quality));
+    /* Perform actual Leiden algorithm iteratively. We either
+     * perform a fixed number of iterations, or we perform
+     * iterations until the quality remains unchanged. Note that
+     * the Leiden algorithm can still change in a subsequent 
+     * iteration. 
```

Comment:
I don't understand the last sentence. Do you mean that even if we use `n_iterations < 0`, another application of the function to the result may produce a change just because the algorithm is randomized?

Could you clarify the comment?

Reply:
Yes, that is correct. Each iteration might find some possible improvement of the quality function. It is possible that a single iteration didn't find an improvement, but that the next iteration finds an improvement again. This is indeed because the algorithm is randomised. Each iteration considers (possibly) slightly different subsets of nodes to consider for moving around to other communities in order to improve the quality function. 

I will clarify the comment.